### PR TITLE
Migrate over more telemetry definitions

### PR DESCRIPTION
**Pull request by hunterwerlla**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: telemetry/definitions/commonDefinitions.json [hunk 1]
```diff
@@ -21,14 +21,15 @@
             "description": "A generic version metadata"
         },
         {
-            "name": "hasException",
-            "type": "boolean",
-            "description": "Determines if the operation had an exception or not"
+            "name": "reason",
+            "type": "string",
```
Comment (bryceitoc9) [inline telemetry/definitions/commonDefinitions.json]: `reason` is typed in VS Code for `createSamApp`, should that be the case here too?: https://github.com/aws/aws-toolkit-vscode/blob/master/src/lambda/commands/createNewSamApp.ts#L58

Comment (hunterwerlla) [reply telemetry/definitions/commonDefinitions.json]: No because this can be used for other things

Comment (bryceitoc9) [reply telemetry/definitions/commonDefinitions.json]: Will those other things have similar reasons/can we just keep bulking up the type?

Comment (hunterwerlla) [reply telemetry/definitions/commonDefinitions.json]: I'd rather not build a huge enum of reasons, this field is more of a get more information field (that could contain IDE specific information) than a field that we aggregate together across IDEs

**Comment by bryceitoc9** [comment]
Please change `result` to `telemetryResult` as `result` is too common of a variable name to be reserved exclusively for telemetry.

File: telemetry/definitions/vscodeDefinitions.json [hunk 2]
```diff
@@ -4,6 +4,57 @@
             "name": "vscode_activeregions",
```
Comment (bryceitoc9) [inline telemetry/definitions/vscodeDefinitions.json]: Nit: `vscode_activeRegions`

Comment (hunterwerlla) [reply telemetry/definitions/vscodeDefinitions.json]: Fixed

File: telemetry/vscode/src/generate.ts [hunk 3]
```diff
@@ -5,11 +5,15 @@
 
 import { MetadataType, MetricMetadataType, MetricMetadata, Metric, MetricDefinitionRoot } from './parser'
 
-// converts snake_case to CamelCase. E.x. lambda_invoke => LambdaInvoke
+function toTitleCase(s: string): string {
+    return s.replace(s[0], s[0].toUpperCase())
```
Comment (bryceitoc9) [inline telemetry/vscode/src/generate.ts]: Nit: Looks like the generally accepted way to do this is
```
return s[0].toUpperCase() + s.slice(1)
```

**Pull request merged**